parcels/grid.py

```diff
@@ -275,22 +297,32 @@ def advancetime(self, grid_new):
         else:
             raise RuntimeError("Time of field_new in Field.advancetime() overlaps with times in old Field")
 
+    def check_zonal_periodic(self):
```

Comment:
Why is this function duplicated? Make it globally accessible?

Reply:
Those functions are actually not the same
`dx = self.lon[0, 1:] - self.lon[0, :-1]`
vs
`dx = self.lon[1:] - self.lon[:-1]`

But since they are almost equal, we should maybe still define the function in the parent class `Grid` and manage the difference with a simple if. 
Do you have a preference for the style? I would put the function in `Grid`

Reply:
Yes, I agree